### PR TITLE
Ben/xdoc

### PR DESCRIPTION
**Pull request by benleetownsend**
TODOs:
* Add activation tests
* Push base model to s3
* Remove some undesired mantis changes.

### CODE REVIEW

File: finetune/base_models/bert/model.py [hunk 1]
```diff
@@ -452,3 +452,56 @@ class LayoutLM(_BaseBert):
...
+            "bottom": 0,
+        },
+        "use_auxiliary_info": True,
+        "low_memory_mode": True,
+        "base_model_path": os.path.join("bert", "XDoc.jl"),
+        "act_fn": "hf_gelu",
+    }
+    required_files = [
+        # {
+        #     "file": os.path.join(
```
Comment (madisonmay) [inline finetune/base_models/bert/model.py:482]: Remove me

File: finetune/base_models/bert/model.py [hunk 2]
```diff
@@ -452,3 +452,56 @@ class LayoutLM(_BaseBert):
...
+class XDocBase(_BaseBert):
+    featurizer = xdoc_featurizer
+    encoder = RoBERTaEncoderXDoc
+    is_roberta = False
+    settings = {
+        **BERT_BASE_PARAMS,
+        "epsilon": 1e-8,
+        "lr": 1e-4,
+        "context_injection": True,
+        "crf_sequence_labeling": False,
```
Comment (madisonmay) [inline finetune/base_models/bert/model.py:467]: Intentionally preventing reading order bias?

File: finetune/encoding/input_encoder.py [hunk 3]
```diff
@@ -298,7 +298,8 @@ def tokenize_context(context, encoded_output, config):
     current_char_loc = 0
     tokenized_context = []
     assert len(encoded_output.tokens) == len(encoded_output.token_ends)
-    assert encoded_output.token_starts[1] <= encoded_output.token_ends[-2]
+    #print(encoded_output.token_starts)
```
Comment (madisonmay) [inline finetune/encoding/input_encoder.py:301]: Remove me

File: finetune/model.py [hunk 4]
```diff
@@ -28,6 +29,7 @@ class PredictMode:
     ASSOCIATION = "ASSOCIATION"
     ASSOCIATION_PROBAS = "ASSOCIATION_PROBA"
     EXPLAIN = "EXPLAIN"
+    PASSTHROUGH = "PASSTHROUGH"
```
Comment (madisonmay) [inline finetune/model.py:32]: Is this still used?

File: finetune/model.py [hunk 5]
```diff
@@ -242,39 +243,40 @@ def _model_fn(features, labels, mode, params):
...
-                        predictions.update(pred_proba_op)
-                    else:
-                        predictions[PredictMode.NORMAL] = pred_op
-                        predictions[PredictMode.PROBAS] = pred_proba_op
-
-                    if build_explain:
-                        predictions[PredictMode.EXPLAIN] = target_model_state[
-                            "explanation"
-                        ]
+                if target_model_state is not None:
```
Comment (madisonmay) [inline finetune/model.py:246]: Ditto -- do we still need this conditional?

**Comment by benleetownsend** [comment]
Closing and comments have been addressed in #744 - If we need a separate PR I can pick that one apart.

**Pull request closed without merging**